src/Collection.php

```diff
@@ -90,11 +93,22 @@ public static function range($start = 0, $end = null, $step = 1)
 
     /**
      * {@inheritdoc}
+     * @throws InvalidReturnValue
      */
     public function getIterator()
     {
         if ($this->inputFactory) {
-            $this->input = call_user_func($this->inputFactory);
+            $input = call_user_func($this->inputFactory);
+
+            if (is_array($input)) {
+                $input = new RecursiveArrayIterator($input);
+            }
+
+            if (!($input instanceof Traversable)) {
```

Comment:
I added the parentheses to stick with the convention used elsewhere in this project (see [here](https://github.com/DusanKasan/Knapsack/blob/31cf23a32dfbb6de3770dae8368fbc8583425824/src/CollectionTrait.php#L730) and [here](https://github.com/DusanKasan/Knapsack/blob/31cf23a32dfbb6de3770dae8368fbc8583425824/src/collection_functions.php#L1399)).